Modules/Core/Common/src/itkObjectFactoryBase.cxx

```diff
@@ -237,9 +238,9 @@ ObjectFactoryBase::Initialize()
 {
   itkInitGlobalsMacro(PimplGlobals);
 
-  if (!m_PimplGlobals->m_Initialized)
+  // Atomically set m_Initialized to true. If it was false before, enter the if.
+  if (m_PimplGlobals->m_Initialized.exchange(true) == false)
```

Comment:
I would prefer to keep the `if(!x)` syntax, rather than `if(x == false)`.

Reply:
The comment before this line explains the logic, so the comparison can have shorter form as Niels suggests.

Reply:
Done.